confighelper/beaconbroadcaster.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2017-2018 Joel Allen Luellwitz and Andrew Klapp
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+""" Provides the broadcasting component of a a filesystem-based IPC mechanism."""
```

Comment:
We might want to put in more detail somewhere about why we went with a filesystem based IPC mechanism as opposed to D-Bus or Unix sockets. (This something I think people are really going to wonder about.)

Reply:
Shouldn't this go in the README or something?

Reply:
No. I think this is something someone will ask when looking at the code. The README should be somewhat focused on building confidence in our system. (For anyone auditing our public code, the README shouldn't be misleading in any way though.)